### PR TITLE
Add calories energy units

### PR DESCRIPTION
**Pull request by dudanov**
## Proposed change

[Discuss](https://github.com/home-assistant/architecture/discussions/1116#discussion-6965863) of this change.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/122796
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Expanded the units of measurement for the **energy** and **energy_storage** device classes, now supporting additional units like joules, kilojoules, and calories.
- **Documentation**
	- Updated sensor documentation to reflect the new units, enhancing clarity and usability for a broader range of energy-related measurements.

### CODE REVIEW

File: source/_integrations/sensor.markdown [hunk 1]
```diff
@@ -46,8 +46,8 @@ The following device classes are supported for sensors:
 - **date**: Date string (ISO 8601)
 - **distance**: Generic distance in km, m, cm, mm, mi, yd, or in
 - **duration**: Duration in d, h, min, or s
-- **energy**: Energy in Wh, kWh, MWh, MJ, or GJ
-- **energy_storage**: Stored energy in Wh, kWh, MWh, MJ, or GJ
+- **energy**: Energy in J, kJ, MJ, GJ, Wh, kWh, MWh, cal, kcal, Mcal or Gcal
```
Comment (c0ffeeca7) [inline source/_integrations/sensor.markdown]:
```suggestion
- **energy**: Energy in J, kJ, MJ, GJ, Wh, kWh, MWh, cal, kcal, Mcal, or Gcal
```

File: source/_integrations/sensor.markdown [hunk 2]
```diff
@@ -46,8 +46,8 @@ The following device classes are supported for sensors:
...
+- **energy_storage**: Stored energy in J, kJ, MJ, GJ, Wh, kWh, MWh, cal, kcal, Mcal or Gcal
```
Comment (c0ffeeca7) [inline source/_integrations/sensor.markdown]:
```suggestion
- **energy_storage**: Stored energy in J, kJ, MJ, GJ, Wh, kWh, MWh, cal, kcal, Mcal, or Gcal
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @dudanov :+1:

**Comment by dudanov** [comment]
Hi @c0ffeeca7
[Parent PR](https://github.com/home-assistant/core/pull/122796) has been merged.

**Pull request merged**